test/test_clients_api.js

```diff
@@ -34,130 +35,134 @@ describe('Clients API', () => {
   });
 
+  beforeEach(() => {
+    return db().then(models => {
+      models.Clients.destroy({ where: {} });
+    });
+  });
+
   describe('POST ' + endpointPrefix + '/clients', () => {
-    it('should respond 401 Unauthorized if there is no auth header', done => {
+    const postSuccess = (done, body) => {
       server.post(endpointPrefix + '/clients')
+            .set('Authorization', 'Bearer ' + token)
+            .send(body)
             .expect('Content-type', /json/)
-            .expect(401)
             .end((err, res) => {
-              res.status.should.be.equal(401);
-              res.body.code.should.be.equal(401);
-              res.body.errno.should.be.equal(errnos[ERRNO_UNAUTHORIZED]);
-              res.body.error.should.be.equal(errors[UNAUTHORIZED]);
+              should.not.exist(err);
+              res.status.should.be.equal(201);
+              res.body.key.should.be.instanceof(String)
+                 .and.have.lengthOf(16);
+              res.body.secret.should.be.instanceof(String)
+                 .and.have.lengthOf(128);
               done();
             });
-    });
+    };
 
-    it('should respond 400 BadRequest if name param is missing', done => {
+    const postError = (done, code, error, errno, body) => {
       server.post(endpointPrefix + '/clients')
             .set('Authorization', 'Bearer ' + token)
+            .send(body)
             .expect('Content-type', /json/)
-            .expect(400)
+            .expect(code)
             .end((err, res) => {
-              res.status.should.be.equal(400);
-              res.body.code.should.be.equal(400);
-              res.body.errno.should.be.equal(errnos[ERRNO_INVALID_API_CLIENT_NAME]);
-              res.body.error.should.be.equal(errors[BAD_REQUEST]);
+              res.status.should.be.equal(code);
+              res.body.code.should.be.equal(code);
+              res.body.errno.should.be.equal(
+                errnos[errno]);
+              res.body.error.should.be.equal(errors[error]);
               done();
             });
-    });
+    };
 
-    it('should respond 400 BadRequest if name param is empty', done => {
+    it('should respond 401 Unauthorized if there is no auth header', done => {
       server.post(endpointPrefix + '/clients')
-            .set('Authorization', 'Bearer ' + token)
-            .send({ name: '' })
             .expect('Content-type', /json/)
-            .expect(400)
+            .expect(401)
             .end((err, res) => {
-              res.status.should.be.equal(400);
-              res.body.code.should.be.equal(400);
-              res.body.errno.should.be.equal(errnos[ERRNO_INVALID_API_CLIENT_NAME]);
-              res.body.error.should.be.equal(errors[BAD_REQUEST]);
+              res.status.should.be.equal(401);
+              res.body.code.should.be.equal(401);
+              res.body.errno.should.be.equal(errnos[ERRNO_UNAUTHORIZED]);
+              res.body.error.should.be.equal(errors[UNAUTHORIZED]);
```

Comment:
We add the auth header within `postError`, adding yet another parameter to `postError` to remove the auth header for a single test didn't seem worthy... But it's not a big deal anyway.